client/src/main/java/com/tigrisdata/db/client/service/AbstractTigrisDBClient.java

```diff
@@ -0,0 +1,56 @@
+package com.tigrisdata.db.client.service;
+
+import com.tigrisdata.db.client.auth.AuthorizationToken;
+import com.tigrisdata.db.client.config.TigrisDBConfiguration;
+import com.tigrisdata.db.client.interceptors.AuthHeaderInterceptor;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Metadata;
+import io.grpc.stub.MetadataUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractTigrisDBClient {
+  protected final ManagedChannel channel;
+
+  private static final Metadata.Key<String> USER_AGENT_KEY =
+      Metadata.Key.of("user-agent", Metadata.ASCII_STRING_MARSHALLER);
+  private static final Metadata.Key<String> CLIENT_VERSION_KEY =
+      Metadata.Key.of("client-version", Metadata.ASCII_STRING_MARSHALLER);
+
+  private static final String USER_AGENT_VALUE = "tigrisdb-client-java.grpc";
+  private static final String CLIENT_VERSION_VALUE = "1.0";
+  private static final Logger log = LoggerFactory.getLogger(AbstractTigrisDBClient.class);
+
+  protected AbstractTigrisDBClient(
+      TigrisDBConfiguration clientConfiguration, AuthorizationToken authorizationToken) {
+
+    ManagedChannelBuilder channelBuilder =
+        ManagedChannelBuilder.forTarget(clientConfiguration.getBaseURL())
+            .intercept(new AuthHeaderInterceptor(authorizationToken))
+            .intercept(MetadataUtils.newAttachHeadersInterceptor(getDefaultHeaders()));
+    if (clientConfiguration.getNetwork().isUsePlainText()) {
+      log.warn(
+          "Client is configured to use plaintext communication. It is advised to not use plaintext communication");
+      channelBuilder.usePlaintext();
+    }
+    this.channel = channelBuilder.build();
+  }
+
+  protected AbstractTigrisDBClient(
+      AuthorizationToken authorizationToken,
+      ManagedChannelBuilder<? extends ManagedChannelBuilder> managedChannelBuilder) {
+    this.channel =
+        managedChannelBuilder
+            .intercept(new AuthHeaderInterceptor(authorizationToken))
+            .intercept(MetadataUtils.newAttachHeadersInterceptor(getDefaultHeaders()))
+            .build();
+  }
+
+  private static Metadata getDefaultHeaders() {
+    Metadata defaultHeaders = new Metadata();
+    defaultHeaders.put(USER_AGENT_KEY, USER_AGENT_VALUE);
+    defaultHeaders.put(CLIENT_VERSION_KEY, CLIENT_VERSION_VALUE);
+    return defaultHeaders;
```

Comment:
this is great, are you thinking of passing the dns name as well in the header? Should we define a const for that as well?

Reply:
added another metadata - we can formalize naming later. Currently this is just sending the DNS name. Once we add multiple peer support, we can fine tune it further.

Reply:
sounds good.